### ISSUE TITLE
Fix S1121 FP: Assignments should not be made from within sub-expressions

### ISSUE THREAD

**Issue by KuraiAndras** [SonarSource/sonar-dotnet#9774]
### Description

With C# 14 [null-conditional assignents](https://learn.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-14#null-conditional-assignment) S1121 is raised, and I think it should not.

### Reproducer

```csharp
var asd = GetMyClass();

asd?.Data?.Length = 5;

Console.WriteLine(asd?.Data?.Length);

static MyClass? GetMyClass()
{
    return new MyClass();
}


class MyClass
{
    public MyData? Data { get; set; }
}

class MyData
{
    public int Length { get; set; }
}
```

### Product and Version

.NET 10, C# 14, SonarAnalyzer.CSharp 10.15.0.120848

**Comment by jilles-sg** [comment]
This appears to be a duplicate of https://github.com/SonarSource/sonar-dotnet/issues/9761 .

**Comment by KuraiAndras** [comment]
Yeah, sorry, didn't see the other one when searching
